app/routes/user/password.tsx

```diff
@@ -0,0 +1,140 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+import type { DataFunctionArgs } from '@remix-run/node'
+import { useFetcher } from '@remix-run/react'
+import { Button, Icon, Label, TextInput } from '@trussworks/react-uswds'
+import { useState } from 'react'
+
+import { PasswordManager } from './password.server'
+import Spinner from '~/components/Spinner'
+import { getFormDataString } from '~/lib/utils'
+
+export const handle = {
+  breadcrumb: 'Password',
+  getSitemapEntries: () => null,
+}
+
+export async function action({ request }: DataFunctionArgs) {
+  const data = await request.formData()
+  const oldPassword = getFormDataString(data, 'oldPassword')
+  const newPassword = getFormDataString(data, 'newPassword')
+  const confirmPassword = getFormDataString(data, 'confirmPassword')
+
+  if (!oldPassword || !newPassword || !confirmPassword) {
+    throw new Response('all password fields must be present', { status: 400 })
+  }
+
+  if (newPassword != confirmPassword) {
+    throw new Response('passwords must match', { status: 400 })
+  } else {
+    await PasswordManager.updatePassword(request, oldPassword, newPassword)
+
+    return null
+  }
+}
+
+export default function () {
+  const fetcher = useFetcher<typeof action>()
+  const checkPassword = (str: string) => {
+    var re = /^(?=.*\d)(?=.*[!@#$%^&*])(?=.*[a-z])(?=.*[A-Z]).{8,}$/
+    return re.test(str)
+  }
+  const startsOrEndsWithWhitespace = (str: string) => {
+    return /^\s|\s$/.test(str)
+  }
+  const [newPassword, setNewPassword] = useState('')
+  const [confirmPassword, setConfirmPassword] = useState('')
+  const [isNewPasswordTouched, setIsNewPasswordTouched] = useState(false)
+  const [isConfirmPasswordTouched, setIsConfirmPasswordTouched] =
+    useState(false)
+  const passwordsMatch = newPassword === confirmPassword
+  const passwordsAreEmpty = newPassword === '' && confirmPassword === ''
```

Comment:
Empty strings are falsey:

```suggestion
  const passwordsAreEmpty = newPassword && confirmPassword
```

Reply:
<img width="310" alt="Screen Shot 2023-07-03 at 3 29 14 PM" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/6545874/437645a4-9fad-441e-8cdd-fd0c43422e60">

Reply:
<img width="327" alt="Screen Shot 2023-07-05 at 8 14 26 AM" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/6545874/5d0de77f-3a10-4f0e-8301-0f57e0927bf7">
it does not work if I do not specify the triple equals empty string.